src/base/rss/rss_feed.cpp

```diff
@@ -110,6 +110,24 @@ void Feed::markAsRead()
     }
 }
 
+void Feed::markAsUnread()
+{
+    auto oldUnreadCount = m_unreadCount;
+    foreach (Article *article, m_articles) {
+        if (article->isRead()) {
+            //article->disconnect(this);
+            article->markAsUnread();
```

Comment:
If the article is being disconnected after being marked read, it would need to be reconnected here.